picocrt/meson.build

```diff
@@ -36,6 +36,17 @@
 src_picocrt = []
 src_picocrt_none = files('crt0-none.c')
 
+# Machine-specific crt0 variants to build.
+picocrt_machines = [
+  {
+    # Must match a valid value of the 'test-machine' option, and will cause the
+    # MACHINE_<name> preprocessor macro to be defined when compiling crt0.
+    'name': 'qemu',
+    # Suffix used on file names, QEMU's is empty because it is the default.
+    'suffix': '',
+  },
+]
+
```

Comment:
(thanks, github for making this a bit harder).

After the machine_dir has been processed, we need to make sure the test_machine value is valid. Something like this would be helpful:
```
diff --git a/picocrt/meson.build b/picocrt/meson.build
index 6d416b1d2..687c4418d 100644
--- a/picocrt/meson.build
+++ b/picocrt/meson.build
@@ -55,6 +55,17 @@ else
   src_picocrt = files('shared/crt0.c')
 endif
 
+machine_found = false
+foreach machine : picocrt_machines
+  if machine['name'] == test_machine
+    machine_found = true
+  endif
+endforeach
+
+if not machine_found
+  error(test_machine + ': requested test machine not found')
+endif
+
```